pkg/server/api_v2_databases_metadata_test.go

```diff
@@ -544,29 +548,122 @@ func TestGetTableMetadataUpdateJobStatus(t *testing.T) {
 		userClient, _, err := ts.GetAuthenticatedHTTPClientAndCookie(sessionUsername, false, 1)
 		require.NoError(t, err)
 
-		failed := makeApiRequest[interface{}](t, userClient, ts.AdminURL().WithPath(uri).String())
+		failed := makeApiRequest[interface{}](t, userClient, ts.AdminURL().WithPath(uri).String(), http.MethodGet)
 		require.Equal(t, http.StatusText(http.StatusNotFound), failed)
 
 		_, e := conn.Exec(fmt.Sprintf("GRANT CONNECT ON DATABASE defaultdb TO %s", sessionUsername.Normalized()))
 		require.NoError(t, e)
 
-		mdResp := makeApiRequest[tmUpdateJobStatusResponse](t, userClient, ts.AdminURL().WithPath(uri).String())
+		mdResp := makeApiRequest[tmUpdateJobStatusResponse](t, userClient, ts.AdminURL().WithPath(uri).String(), http.MethodGet)
 		require.Equal(t, "NOT_RUNNING", mdResp.CurrentStatus)
 
 		_, e = conn.Exec(fmt.Sprintf("REVOKE CONNECT ON DATABASE defaultdb FROM %s", sessionUsername.Normalized()))
 		require.NoError(t, e)
-		failed = makeApiRequest[string](t, userClient, ts.AdminURL().WithPath(uri).String())
+		failed = makeApiRequest[string](t, userClient, ts.AdminURL().WithPath(uri).String(), http.MethodGet)
 		require.Equal(t, http.StatusText(http.StatusNotFound), failed)
 
 		_, e = conn.Exec(fmt.Sprintf("GRANT admin TO %s", sessionUsername.Normalized()))
 		require.NoError(t, e)
-		mdResp = makeApiRequest[tmUpdateJobStatusResponse](t, userClient, ts.AdminURL().WithPath(uri).String())
+		mdResp = makeApiRequest[tmUpdateJobStatusResponse](t, userClient, ts.AdminURL().WithPath(uri).String(), http.MethodGet)
 		require.Equal(t, "NOT_RUNNING", mdResp.CurrentStatus)
 	})
 }
 
-func makeApiRequest[T any](t *testing.T, client http.Client, uri string) (mdResp T) {
-	req, err := http.NewRequest("GET", uri, nil)
+func TestTriggerMetadataUpdateJob(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	jobCompletedChan := make(chan interface{})
+	jobReadyChan := make(chan interface{})
+	defer close(jobCompletedChan)
+	skip.UnderStress(t, "too slow under stress")
+	testCluster := serverutils.StartCluster(t, 3, base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+
+			Knobs: base.TestingKnobs{
+				TableMetadata: &tablemetadatacache_util.TestingKnobs{
+					OnJobReady: func() {
+						jobReadyChan <- struct{}{}
+					},
+					OnJobComplete: func() {
+						jobCompletedChan <- struct{}{}
+					},
+				},
+			},
+		},
+	})
+	ctx := context.Background()
+	defer testCluster.Stopper().Stop(ctx)
+	conn := testCluster.ServerConn(0)
```

Comment:
If the queries executed in this connection should always succeed, you can wrap this in `sqlutils.MakeSQLRunner` so you don't have to check the error every time.

Reply:
Done